### DISCUSSION TITLE
SQLCipher Integration

### DISCUSSION THREAD

**Discussion by tearfulDalvik** [vlcn-io/cr-sqlite#400] [Ideas]
Hello,

Is there a way to integrate with sqlcipher(https://github.com/sqlcipher/sqlcipher) ?

Thanks!

**Comment by tantaman** [comment]
It should just work given cr-sqlite avoids modifying low level details of sqlite as much as possible.

I'd try loading the extension in sqlcipher and see what happens. If you do, can you let me know the results?

The areas where it could break:
- cr-sqlite requires exclusive use of the `commit`, `update`, and `rollback` hook. If sqlcipher or some other extension or app uses those, it won't work

**Reply by devon** [reply]
It seems like they do use callbacks, check `sqlite3_rollback_hook` in https://github.com/sqlcipher/sqlcipher/blob/7c460791eba939e6c6872825219a6644ca47283b/src/main.c#L2304.

Can we chain the callbacks in any way?

**Comment by tearfulDalvik** [comment]
Thanks for your attention :)
